pkg/auth/webapp/auth_entry_point_middleware.go

```diff
@@ -23,7 +24,7 @@ func (m AuthEntryPointMiddleware) Handle(next http.Handler) http.Handler {
 		}
 
 		if userID != nil && !fromAuthzEndpoint {
-			defaultRedirectURI := "/settings"
+			defaultRedirectURI := DefaultPostLoginRedirectURI(m.UIConfig)
```

Comment:
Just thinking if developer inputs the authgear url into the DefaultPostLoginRedirectURI, it will cause too many redirect here. Do you think it is necessary to check and avoid developer to have such config?

Reply:
To detect infinite redirection, we need to maintain a separate list of routes that will go through AuthEntryPointMiddleware, in addition to adding the routes in `pkg/auth/routes.go`. We also need to parse the redirect URI. First we need to check if the host is custom domain or default domain. If it is neither, then there is properly no infinite redirection. However, it is possible that the redirection first goes to `https://myapp.com/redirect` and then `https://myapp.authgearapps.com/login`, which we can only detect by introducing some state to keep track of the redirection.